Add support for Privacy Regulations | Fix video param validation

shubhamc-ins opened the pull request:
- dsa
- us_privacy/ccpa
- gpp
- gppSid
- coppa

shubhamc-ins committed:
add support for:
- dsa
- us_privacy/ccpa
- gpp
- gppSid

shubhamc-ins committed:
video validation fix

rohanInsticator commented on modules/insticatorBidAdapter.js:
```diff
@@ -190,17 +205,59 @@ function buildDevice(bidRequest) {
 }
 
+function _getCoppa() {
+  return config.getConfig('coppa') === true ? 1 : 0;
```
whose config does this refer?

shubhamc-ins replied:
This is the common config from the prebid module.

rohanInsticator commented on modules/insticatorBidAdapter.js:
```diff
@@ -190,17 +205,59 @@ function buildDevice(bidRequest) {
+}
+
+function _getGppConsent(bidderRequest) {
+  let gpp = deepAccess(bidderRequest, 'gppConsent.gppString')
```
I thought all privacy policy values were available by default from prebid js core module. Why do we need to rely on the publisher's bidderRequest object?

shubhamc-ins replied:
No, it comes from the bidderRequest. 
This is how all other adaptors are set up as well.

shubhamc-ins replied:
@rohanInsticator Check this https://docs.prebid.org/dev-docs/bidder-adaptor.html#bidderrequest-parameters

shubhamc-ins committed:
support first part site details

shubhamc-ins committed:
support user data from bidder, and update server endpoint for alternative bidder configs

rohanInsticator commented on modules/insticatorBidAdapter.js:
```diff
@@ -12,35 +12,53 @@ const USER_ID_COOKIE_EXP = 2592000000; // 30 days
+  'search': v => typeof v === 'string',
+  'mobile': v => isInteger(),
+  'content': v => typeof v === 'object',
+  'keywords': v => typeof v === 'string',
```
even though this is under site in prebid js, we can use this for both site & content in bid requests on the exchange right?

shubhamc-ins replied:
Yes, correct!

shubhamc-ins replied:
Also we are not doing any validation on `content` object from here. We can do that in exchange. @rohanInsticator

rohanInsticator reviewed (commented):
how about another test-case if only one or more of the privacy policies are present?

And also, something more for video changes?

shubhamc-ins committed:
- enhance user object with ext and consent
- remove video param validation

shubhamc-ins committed:
add test case for more than one privacy platforms

shubhamc-ins commented:
> how about another test-case if only one or more of the privacy policies are present?
> 
> And also, something more for video changes?

I added few more test cases and updated the PR. Please check now. @rohanInsticator

rohanInsticator commented on modules/insticatorBidAdapter.js:
```diff
@@ -190,31 +197,96 @@ function buildDevice(bidRequest) {
+  let regs = {
+    ext: {},
+  };
   if (bidderRequest.gdprConsent) {
```
Check how rubicon adapter works, they pass consent string regardless of gdpr consent object is true or not. We should do that too since the consentString can be there for some Non-EU countries too. Please update the same for user object too

shubhamc-ins replied:
Done! @rohanInsticator

shubhamc-ins replied:
Actually, if you see the latest code of rubicon.
![Screenshot 2024-03-26 at 11 28 18 AM](https://github.com/Insticator/Prebid.js/assets/127132399/c6ac73bd-a605-4085-a080-80459880dc3b)

They do check for `gdprConsent` and have removed the `user.ext.consent` as `regs.ext.gdprConsentString` servers the same purpose.
@rohanInsticator

rohanInsticator replied:
User consent is there in rubicon, not sure what you were checking

utils.deepSetValue(data, 'user.ext.consent', bidderRequest.gdprConsent.consentString);

shubhamc-ins replied:
@rohanInsticator you are checking our repository, it's master is not synced with Prebid's official master branch.

Check this https://github.com/prebid/Prebid.js/blob/master/modules/rubiconBidAdapter.js

rohanInsticator replied:
I was looking at PrebidJS repo only in my intellij. I cannot pull the latest on master over there somehow. Even if rubicon has removed it, they must be appending it on the Server Side as IAB said it's good to have that field inside User Object too.

shubhamc-ins committed:
update user.ext.consent

shubhamc-ins committed:
update coppa and support video context
